### DISCUSSION TITLE
Commitizen support

### DISCUSSION THREAD

**Discussion by viocost** [magit/magit#4470] [Support] [answered]
Is there a way to use magit with commitizen?

**Comment by tarsius** [comment, marked as answer]
I assume we are talking about https://github.com/commitizen/cz-cli.

No, this does not work. `git cz` needs a "smart" terminal for the user to input their choices and even move around in the output (as I concluded from looking at the screenshot on that page). Magit doesn't run `git` in a terminal, it doesn't even invoke a shell. Instead it calls `git` directly.

You could however type `git cz` into a terminal, answer the interactive questions and then when `git cz` finally invokes the `$EDITOR`, that could be `emacsclient` configure to connect to the running `emacs` instance. You either have to use magit in the instance before doing this or have `(require 'git-commit)` in you init file. In either case emacs must already be running.
